### PR TITLE
tracy: 0.10 -> 0.11.1 + other changes

### PR DESCRIPTION
**Pull request by paveloom**
## Description of changes

https://github.com/wolfpld/tracy/releases/tag/v0.11.0
https://github.com/wolfpld/tracy/releases/tag/v0.11.1

Supersedes https://github.com/NixOS/nixpkgs/pull/320273. A follow up to https://github.com/NixOS/nixpkgs/pull/315205.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/tr/tracy/default.nix [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    ++ lib.optionals (stdenv.isDarwin || !withWayland) [ glfw ]
+    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.AppKit ]
+    ++ lib.optionals (stdenv.isDarwin && lib.versionAtLeast stdenv.hostPlatform.darwinMinVersion "11") [
+      darwin.apple_sdk.frameworks.UniformTypeIdentifiers
+    ];
+
+  cmakeFlags = [
+    "-DDOWNLOAD_CAPSTONE=off"
+    "-DTRACY_STATIC=off"
+  ] ++ lib.optional (stdenv.isLinux && !withWayland) "-DLEGACY=on";
```
Comment (EzequielRamis) [inline pkgs/by-name/tr/tracy/default.nix:67]: This flag let me open the native file selector on linux using GTK. Currently it fails.
```nix
++ lib.optional stdenv.isLinux "-DGTK_FILESELECTOR=ON";
```

File: pkgs/by-name/tr/tracy/default.nix [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  patches = lib.optional (
+    stdenv.isDarwin && lib.versionOlder stdenv.hostPlatform.darwinMinVersion "11"
+  ) ./dont-use-the-uniformtypeidentifiers-framework.patch;
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+  ] ++ lib.optionals stdenv.cc.isClang [ stdenv.cc.cc.libllvm ];
```
Comment (EzequielRamis) [inline pkgs/by-name/tr/tracy/default.nix:43]: Note that `gtk3` is another requirement on linux (https://github.com/wolfpld/tracy/pull/94).
```nix
++ lib.optionals stdenv.isLinux [ gtk3 ];
```

File: pkgs/by-name/tr/tracy/default.nix (hunk 1 above)
Comment (paveloom) [reply pkgs/by-name/tr/tracy/default.nix]: It works fine for me on GNOME (the GTK4 file chooser is opened). Seems like you are missing a [backend](https://wiki.archlinux.org/title/XDG_Desktop_Portal#List_of_backends_and_interfaces) for the XDG Desktop Portal.

Added an option to use the GTK file selector to the Nix derivation. Note that it is `false` by default, as that's what upstream [expects](https://github.com/wolfpld/tracy/blob/v0.11.0/profiler/CMakeLists.txt#L4).

**Review by JasmineLowen — APPROVED** [review]
I was just about to write the update when I spotted yours. I build it on my machine (`tracy-wayland` on `x86_64-linux`) and it works fine. Code also looks fine in my eyes but I'm not an nix expert either (!)

Thanks for this PR!

**Comment by carlthome** [comment]
There's some conflicts blocking a merge, I assume. Would be nice to have this in!

**Review by shiryel — APPROVED** [review]

**Comment by jacereda** [comment]
Thanks for working on this @paveloom 

I've checked your changes, the conflicts are due to the `stdenv.isXXX -> stdenv.hostPlatform.isXXX` treewide rename. Just changing those will fix it. Tested it and it works fine. Could you rebase, please?

**Comment by Mrmaxmeier** [comment]
LGTM and works nicely on Wayland (Plasma).
In case this is useful: To test I've rebased your changes and resolved conflicts. (Looks like packages were formatted tree-wide recently)
Feel free to use/take these rebased commits :slightly_smiling_face:  https://github.com/NixOS/nixpkgs/compare/master...Mrmaxmeier:nixpkgs:tracy-wayland-0.11.1

**Comment by flokli** [comment]
Thanks. This also fixes the build with the current boost version.

**Pull request merged**